File: rust/tests/parse_ruby_test.rs
use herb::parse_ruby;
use insta::assert_snapshot;

#[test]
fn test_parse_ruby_class() {
  let result = parse_ruby("class Foo; end").unwrap();
  assert_snapshot!(result.prettyprint());
}

#[test]
fn test_parse_ruby_method_definition() {
  let result = parse_ruby("def greet(name)\n  \"Hello, #{name}!\"\nend").unwrap();
  assert_snapshot!(result.prettyprint());
}

#[test]
fn test_parse_ruby_simple_expression() {
  let result = parse_ruby("1 + 2").unwrap();
  assert_snapshot!(result.prettyprint());
}

File: javascript/packages/node/test/parse-ruby.test.ts
import { describe, test, expect, beforeAll } from "vitest"
import { Herb, inspectPrismNode } from "../src/index.ts"

describe("parseRuby", () => {
  beforeAll(async () => {
    await Herb.load()
  })

  test("parseRuby() parses a simple expression", () => {
    const source = "1 + 2"
    const result = Herb.parseRuby(source)
    expect(inspectPrismNode(result.value, source)).toMatchSnapshot()
  })

  test("parseRuby() parses a class definition", () => {
    const source = "class Foo; end"
    const result = Herb.parseRuby(source)
    expect(inspectPrismNode(result.value, source)).toMatchSnapshot()
  })

  test("parseRuby() parses a method definition", () => {
    const source = 'def greet(name)\n  "Hello, #{name}!"\nend'
    const result = Herb.parseRuby(source)
    expect(inspectPrismNode(result.value, source)).toMatchSnapshot()
  })

  test("parseRuby() parses raw Ruby, not ERB", () => {
    const source = "x = 1 + 2"
    const result = Herb.parseRuby(source)
    expect(inspectPrismNode(result.value, source)).toMatchSnapshot()
  })
})
